.ci/do_release.sh

```diff
@@ -1,77 +1,72 @@
 #!/bin/bash -Exel
 
-echo -e "\n\n**********************************"
-echo -e "\n\nStarting do_release.sh script...\n\n"
-echo -e "**********************************\n\n"
+echo "**********************************"
+echo "Starting do_release.sh script..."
+echo "**********************************"
 
-if [ -z "$1" ]; then
-    if [ -z "${release_folder}" ]; then
-        echo "ERROR: Please use the first script argument or env var 'release_folder'. Exit"
-    fi
-else
-    release_folder=$1
+if [ -z "${release_folder}" ]; then
+    echo "ERROR: 'release_folder' was not set. Exit"
+    exit 1
 fi
+
 if [ ! -e "${release_folder}" ] || [ ! -d "${release_folder}" ]; then
     echo "ERROR: [${release_folder}] directory doesn't exist. Exit"
     exit 1
 fi
 
-if [ -z "$2" ]; then
-    if [ -z "${release_version}" ]; then
-        echo "ERROR: Please use the second script argument or env var 'release_version'. Exit"
-    fi
-else
-    release_version=$2
-    echo "FULL_VERSION from script parameter: [${release_version}]"
+if [ -z "${release_tag}" ]; then
+    echo "ERROR: 'release_tag' was not set. Exit"
+    exit 1
+fi
+
+if [ -z "${revision}" ]; then
+    echo "WARN: 'revision' was not set, defaulting to 1"
+    revision=1
 fi
 
-env PRJ_RELEASE=1 contrib/build_pkg.sh -s
+if [ -z "${do_release}" ]; then
+    echo "WARN: 'do_release' was not set, defaulting to false (package will not be release)"
+    do_release=false
+fi
+
+env PRJ_RELEASE="${revision}" contrib/build_pkg.sh -s
 
-MAJOR_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_major\]" | awk -e '{ printf $2 };' | sed  's/)//g')
-MINOR_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_minor\]" | awk -e '{ printf $2 };' | sed  's/)//g')
-REVISION_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_revision\]" | awk -e '{ printf $2 };' | sed  's/)//g')
+MAJOR_VERSION=$(grep -e "define(\[prj_ver_major\]" configure.ac | awk -e '{ printf $2 };' | sed  's/)//g')
```

Comment:
IMO this approach is not bullet proof

You expect some format of version string, what will happened if it's different?

IMO you should add `set -o pipefail` at the beginning of the script to handle failed piped commands

Reply:
The `-e` in `awk` command is redundant here, it serves no function